### PR TITLE
Add System.Net.Http.Native shim and seed it with curl initialization

### PR DESCRIPTION
**Pull request by nguerrera**
cc @sokket @bartonjs @stephentoub

### CODE REVIEW

**Comment by stephentoub** [comment]
cc: @vijaykota, @kapilash

File: src/Common/src/Interop/Unix/System.Net.Http.Native/Interop.Initialization.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography;
```
Comment (nguerrera) [inline src/Common/src/Interop/Unix/System.Net.Http.Native/Interop.Initialization.cs]: Unnecessary using copied and pasted. Could cause problems when included in other projects. Will fix.

File: src/System.Net.Http/src/System.Net.Http.csproj [hunk 2]
```diff
@@ -180,6 +180,13 @@
     <Compile Include="$(CommonPath)\Interop\Unix\libcurl\Interop.SafeCurlHandle.cs">
       <Link>Common\Interop\Unix\libcurl\Interop.SafeCurlHandle.cs</Link>
     </Compile>
+    <Compile Include="$(CommonPath)\Interop\Unix\System.Net.Http.Native\Interop.Initialization.cs">
+      <Link>Common\Interop\Unix\System.Net.Http.Native\Interop.Initialization.cs</Link>
+    </Compile>
+     <Compile Include="$(CommonPath)\Interop\Unix\System.Security.Cryptography.Native\Interop.Initialization.cs">
+      <Link>Common\Interop\Unix\System..Security.Cryptography.Native\Interop.Initialization.cs</Link>
+    </Compile>
+
```
Comment (nguerrera) [inline src/System.Net.Http/src/System.Net.Http.csproj]: Accidental extra whitespace.

File: src/System.Net.Http/src/System.Net.Http.csproj [hunk 3]
```diff
@@ -180,6 +180,13 @@
     <Compile Include="$(CommonPath)\Interop\Unix\libcurl\Interop.SafeCurlHandle.cs">
       <Link>Common\Interop\Unix\libcurl\Interop.SafeCurlHandle.cs</Link>
     </Compile>
+    <Compile Include="$(CommonPath)\Interop\Unix\System.Net.Http.Native\Interop.Initialization.cs">
+      <Link>Common\Interop\Unix\System.Net.Http.Native\Interop.Initialization.cs</Link>
+    </Compile>
+     <Compile Include="$(CommonPath)\Interop\Unix\System.Security.Cryptography.Native\Interop.Initialization.cs">
+      <Link>Common\Interop\Unix\System..Security.Cryptography.Native\Interop.Initialization.cs</Link>
```
Comment (stephentoub) [inline src/System.Net.Http/src/System.Net.Http.csproj]: Nit: double period between System and Security

File: src/Native/System.Net.Http.Native/pal_curlinit.h [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+#pragma once
+
+#include <stdint.h>
+
+/**
+ * Initializes curl. 
+ *
+ * Thread-safe and idempotent. Must be called before using any other curl function.
+ * EnsureOpenSSLIsInitialized from System.Security.Cryptography.Native must already have
+ * have been called before calling this.
```
Comment (stephentoub) [inline src/Native/System.Net.Http.Native/pal_curlinit.h]: Nit: have have

**Comment by stephentoub** [comment]
LGTM

**Comment by bartonjs** [comment]
@dotnet-bot test this please (and leave out the intermittent failures, s'il vous plait)

**Comment by bartonjs** [comment]
LGTM

**Comment by vijaykota** [comment]
LGTM

**Pull request merged**